draft-ietf-httpbis-messaging-latest.xml

```diff
@@ -703,29 +703,22 @@ Host: www.example.org:8001
    the target URI's scheme is "https"; if not, the scheme is "http".
 </li>
 <li>
-   If the server's configuration (or outbound gateway) provides a fixed URI
-   <x:ref>authority</x:ref> component, that authority is used for the
-   target URI. If not, then if the request-target is in
-   <x:ref>authority-form</x:ref>, the target URI's authority
-   component is the same as the request-target.
-   If not, then if a <x:ref>Host</x:ref> header field is supplied with a
-   non-empty field value, the authority component is the same as the
-   Host field value. Otherwise, the authority component is assigned
-   the default name configured for the server and, if the connection's
-   incoming TCP port number differs from the default port for the target
-   URI's scheme, then a colon (":") and the incoming port number (in
-   decimal form) are appended to the authority component.
+   If the request-target is in <x:ref>authority-form</x:ref>,
+   the request-target is the target URI's authority component.
+   Otherwise, the target URI's authority component is the field value of the
+   <x:ref>Host</x:ref> header field, if that value is valid and non-empty, or
+   the authority component is empty if Host is absent, empty, or invalid.
```

Comment:
I think that three sentences works better, but our language doesn't do ternary conditions very well.

```suggestion
   Otherwise, the target URI's authority component is the field value of the
   <x:ref>Host</x:ref> header field, if that value is valid and non-empty.
   In all other cases, the authority component is empty if <x:ref>Host<x:ref>
   is absent, empty, or invalid.
```

or even

```suggestion
   For other request forms, if the <x:ref>Host</x:ref> header field is valid
   and non-empty, the target URI's authority component is that value.
   Otherwise, the authority component is empty if <x:ref>Host<x:ref> is
   absent, empty, or invalid.
```

or, to make the construction more parallel, though I don't know how to say "empty is the authority component" without it seeming like a bad Yoda impression


```suggestion
   For other request forms, if the <x:ref>Host</x:ref> header field is valid
   and non-empty, that value is the target URI's authority component.
   Otherwise, the authority component is empty if <x:ref>Host<x:ref> is
   absent, empty, or invalid.
```

Reply:
I used a fourth option in https://github.com/httpwg/http-core/pull/802/commits/69dbe8ff030a94c9183e9ec023d6c4fb241eb51a